fc/parsing/CompactSyntaxParser.py

```diff
@@ -1481,43 +1498,50 @@ class CompactSyntaxParser(object):
                       p.Combine('MathML:' + p.oneOf(' '.join(mathmlOperators))))
     wrap = p.Group(p.Suppress('@') - Adjacent(p.Word(p.nums)) + Adjacent(colon) + mathmlOperator
                    ).setName('WrapMathML').setParseAction(Actions.Wrap)
-    
+
     # Turning on tracing for debugging protocols
     trace = Adjacent(p.Suppress('?'))
-    
+
     # The main expression grammar.  Atoms are ordered according to rough speed of detecting mis-match.
     atom = (array | wrap | number.copy().setParseAction(Actions.Number) | stringValue |
             ifExpr | nullValue | defaultValue | lambdaExpr | functionCall | identAsVar | tuple).setName('Atom')
-    expr << p.operatorPrecedence(atom, [(accessor, 1, p.opAssoc.LEFT, Actions.Accessor),
-                                        (viewSpec, 1, p.opAssoc.LEFT, Actions.View),
-                                        (index, 1, p.opAssoc.LEFT, Actions.Index),
-                                        (trace, 1, p.opAssoc.LEFT, Actions.Trace),
-                                        ('^', 2, p.opAssoc.LEFT, Actions.Operator),
-                                        ('-', 1, p.opAssoc.RIGHT, lambda *args: Actions.Operator(*args, rightAssoc=True)),
-                                        (p.oneOf('* /'), 2, p.opAssoc.LEFT, Actions.Operator),
-                                        (p.oneOf('+ -'), 2, p.opAssoc.LEFT, Actions.Operator),
-                                        (p.Keyword('not'), 1, p.opAssoc.RIGHT, lambda *args: Actions.Operator(*args, rightAssoc=True)),
-                                        (p.oneOf('== != <= >= < >'), 2, p.opAssoc.LEFT, Actions.Operator),
-                                        (p.oneOf('&& ||'), 2, p.opAssoc.LEFT, Actions.Operator)
-                                       ])
-    
+    expr <<= p.infixNotation(atom, [(accessor, 1, p.opAssoc.LEFT, Actions.Accessor),
```

Comment:
Yikes

Reply:
Is there some way to write this that makes it more obvious what's going on?

Reply:
Not easily! It's not too bad if you're familiar with the pyparsing approach to parsing infix notation. Basically, as the comment 2 lines up says, this is the main grammar for parsing expressions in the post-processing language. So it has to be fairly complex.